test-suite/src/test/scala/org/scalajs/testsuite/jsinterop/ExportsTest.scala

```diff
@@ -969,6 +1005,53 @@ object ExportsTest extends JasmineTest {
       expect(obj2.x).toBe(100)
     }
 
+    it("should offer auto exports for classes extending a trait with ignoreInvalidDescendants") {
+      val ctor =
+        js.Dynamic.global.org.scalajs.testsuite.jsinterop.AutoExportIgnoreTraitClass
+      expect(ctor).toBeDefined
+      expect(js.typeOf(ctor)).toEqual("function")
+
+      val obj1 = js.Dynamic.newInstance(ctor)()
+      expect(obj1).toBeDefined
+      expect(obj1.x).toBe(5)
+
+      val obj2 = js.Dynamic.newInstance(ctor)(100)
+      expect(obj2).toBeDefined
+      expect(obj2.x).toBe(100)
+    }
+
+    it("should offer auto exports for classes extending a class with ignoreInvalidDescendants") {
+      val ctor =
+        js.Dynamic.global.org.scalajs.testsuite.jsinterop.AutoExportIgnoreClassClass
+      expect(ctor).toBeDefined
+      expect(js.typeOf(ctor)).toEqual("function")
+
+      val obj1 = js.Dynamic.newInstance(ctor)()
+      expect(obj1).toBeDefined
+      expect(obj1.x).toBe(5)
+
+      val obj2 = js.Dynamic.newInstance(ctor)(100)
+      expect(obj2).toBeDefined
+      expect(obj2.x).toBe(100)
+    }
+
+    it("should ignore invalid descendants") {
+      // This is just to check that everything here compiles
+      class A extends AutoExportIgnoreTrait { def foo = 1 }
+      class B extends AutoExportIgnoreClass { def foo = 2 }
+
+      val a = new A { override def foo = 3 }
+      val b = new B { override def foo = 4 }
+      val c = new AutoExportIgnoreClass { def bar = 1 }
+      val d = new AutoExportIgnoreTrait { def bar = 1 }
+
+      // Check the classes are usable
+      expect((new A).foo).toEqual(1)
+      expect((new B).foo).toEqual(2)
+      expect(a.foo).toEqual(3)
+      expect(b.foo).toEqual(4)
+    }
```

Comment:
I didn't manipulate them on purpose, since that would be reflective calls... Do you nevertheless want me to do it?